executable/SummarizeApp.re

```diff
@@ -10,55 +10,298 @@ let mockedSummary: User.summary = {
-        field(
-          "projects",
-          ~doc="Total of projects",
-          ~typ=non_null(int),
-          ~args=Arg.[],
-          ~resolve=(info, p: User.summary) =>
-          p.projects
-        ),
-        field(
-          "organizations",
-          ~doc="Total of organizations",
-          ~typ=non_null(int),
-          ~args=Arg.[],
-          ~resolve=(info, p: User.summary) =>
-          p.organizations
-        ),
-      ]
+  User.Summary.(
+    Schema.(
+      obj("summary", ~doc="User's summary of notifications", ~fields=_ =>
+        [
+          field(
+            "unread",
+            ~doc="Count of unread items",
+            ~typ=non_null(int),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.unread
+          ),
+          field(
+            "total",
+            ~doc="Total of notifications",
+            ~typ=non_null(int),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.total
+          ),
+          field(
+            "projects",
+            ~doc="Total of projects",
+            ~typ=non_null(int),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.projects
+          ),
+          field(
+            "organizations",
+            ~doc="Total of organizations",
+            ~typ=non_null(int),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.organizations
+          ),
+        ]
+      )
+    )
+  );
+
+let message =
+  User.Message.(
+    Schema.(
+      obj("message", ~doc="User notification message", ~fields=_ =>
+        [
+          field(
+            "id",
+            ~doc="Unique identifier of message",
+            ~typ=non_null(int),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.id
+          ),
+          field(
+            "email",
+            ~doc="Unique identifier of message",
+            ~typ=non_null(string),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.email
+          ),
+          field(
+            "message",
+            ~doc="Unique identifier of message",
+            ~typ=non_null(string),
+            ~args=Arg.[],
+            ~resolve=(info, p) =>
+            p.message
+          ),
+        ]
+      )
+    )
+  );
+
+let messagesConnectionEdge =
+  Connection.(
+    Schema.(
+      obj("messagesConnectionEdge", ~doc="Message Connection Edge", ~fields=_ =>
+        [
+          field(
+            "cursor",
+            ~doc="Cursor of connection edge",
+            ~typ=non_null(string),
+            ~args=Arg.[],
+            ~resolve=(info, p: edge(User.Message.t)) =>
+            p.cursor
+          ),
+          field(
+            "node",
+            ~doc="Node of connection edge",
+            ~typ=non_null(message),
+            ~args=Arg.[],
+            ~resolve=(info, p: edge(User.Message.t)) =>
+            p.node
+          ),
+        ]
+      )
+    )
+  );
+
+let messagesConnection =
```

Comment:
nice we can probably mount a helper after to help boot the connection structure

Reply:
I just had an idea here, gonna push soon